### PR TITLE
DOC: Add illustration to Figure.subplots_adjust

### LINKED ISSUE #23005: [Doc]: Add figure to `subplots_adjust`
### Documentation Link

https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots_adjust.html

### Problem

"Inspired" by https://github.com/matplotlib/cheatsheets/issues/109

A figure would help understanding the different parameters

### Suggested improvement

Add a figure looking similar to the one in the cheatsheet, but correct and only containing the argument parameters.

### PR DESCRIPTION
**Pull request by timhoffm**
Closes #23005.

Example: https://output.circle-artifacts.com/output/job/227e8182-1813-47e4-8cfc-1689902c5e28/artifacts/0/doc/build/html/api/_as_gen/matplotlib.figure.Figure.subplots_adjust.html#matplotlib.figure.Figure.subplots_adjust

### CODE REVIEW

**Comment by jklymak** [comment]
This is nice but same concern as previous one of these: most folks consume the api docs in IDEs and may be frustrated they cannot reach this.  It'd be nice to work on a solution that allowed both the image and a link.

**Comment by timhoffm** [comment]
[Same answer](https://github.com/matplotlib/matplotlib/pull/28808#issuecomment-2348254367): Link sounds nice but in a text-based renderer a usable realization is non-trivial. How should the link look like? Where should it point to? How would it be used?

The plots I add are supplementary illustrations. You can get all information from the text. It think we have to accept that some interfaces to the docstrings are just less capable and cannost show illustrations / have links. - We also live with the fact that code links like `.Axes.set_xlim` do not work in plain text.

**Comment by jklymak** [comment]
It would take a new directive I suppose, but something like

```
.. example-plot:: https://matplotlib.org/stable/example_plots/example_subplots_adjust.png

```

which then either just display that in the rendered docs, or generated it from `example_plots/example_subplots_adjust.py` would be nicer.  Of course the link being clickable would depend on the IDE.

```
.. image:: https://matplotlib.org/stable/example_plots/example_subplots_adjust.png
```
would be fine too if sphinx allowed the image directive to use an html.  That has a bootstrapping problem in that the example plot would need to be on stable, whereas a new directive that created the file and put in the right spot would be better (the absolute path would be ignored in rendering).

**Comment by timhoffm** [comment]
I would support such a directive. As you said, there are a number of aspects to work out. Related: It may also be an idea to have a link in every API doc something like: "Online version: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.margins.html" at the bottom. With that one gets access not only to the image but cross-references, related examples etc.

But all of this is off-topic. In this PR I'm just adding content, that is for now helpful in the HTML docs. Whether one can additionally find better representations in the raw docstring is orthogonal. Also, I believe it's helpful to start with the content. If we only need/have such images in one or two places, the work for a extension may not be justifed. But having already several plots can be a motivation for the extension.

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1322,6 +1322,10 @@ def subplots_adjust(self, left=None, bottom=None, right=None, top=None,
         Unset parameters are left unmodified; initial values are given by
         :rc:`figure.subplot.[name]`.
 
+        The following plot illustrates the effect of the parameters:
+
```
Comment (jklymak) [inline lib/matplotlib/figure.py:1326]:
```suggestion
```

Suggest you just remove this for now.  It should be obvious what the plot is for, and is less distracting for folks who can't see the plot.

Comment (timhoffm) [reply lib/matplotlib/figure.py]: ok

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Feel free to self merge when build done

**Pull request merged**